packages/web/src/pages/upload-page/UploadPageNew.tsx

```diff
@@ -64,20 +86,63 @@ export const UploadPageNew = (props: UploadPageProps) => {
         <EditPageNew
           tracks={tracks}
           setTracks={setTracks}
-          onContinue={() => setPhase(Phase.FINISH)}
+          onContinue={() => {
+            setPageTitle(
+              tracks.length > 1
+                ? messages.finishMultiTrackPageTitle
+                : messages.finishSingleTrackPageTitle
+            )
+            setPhase(Phase.FINISH)
+            setReadyToUpload(true)
+          }}
         />
       )
       break
     case Phase.FINISH:
-      console.log(tracks[0])
-      page = <pre>{JSON.stringify(tracks, null, 2)}</pre>
+      page = (
+        <FinishPageNew
+          tracks={tracks}
+          onContinue={() => {
+            setTracks([])
+            setPageTitle(messages.selectPageTitle)
+            setReadyToUpload(false)
+            setPhase(Phase.SELECT)
+          }}
+        />
+      )
   }
+
+  const handleUpload = useCallback(() => {
```

Comment:
would it makes sense for this to be a part of the finish page?

Reply:
seems like it could be fine in either place. put it here bc thats how it existed in the legacy upload flow